### DISCUSSION TITLE
App getting terminated in iPhone Devices due to memory issue.

### DISCUSSION THREAD

**Discussion by chaudharyvikram** [airbnb/lottie-ios#1993] [General]
## Which Version of Lottie are you using?

Lottie 4.1.3

## Expected Behavior 

Animation should work smoothly

## Actual Behavior

App getting crashed due to memory issue. In Simulator it works fine. even other animations are working fine but some of them are not working. 
<img width="1792" alt="Screenshot 2023-03-07 at 7 34 48 PM" src="https://user-images.githubusercontent.com/39688232/223455845-ac5c4aab-2e28-420d-b5a9-a487965b9f42.png">

## Animation JSON

Due to privacy reason I can't able to attach json file. Will try to attach after permission.

**Comment by calda** [comment]
Please share a copy of your animation json if you're able to

**Comment by chaudharyvikram** [comment]
Here is Animation file:-  https://drive.google.com/file/d/1p81zap5WqVB35IeFCzoGyc7Ak5zoC7Fp/view?usp=sharing
Please reuest us to access or share email on which I can share animation.
I tested this on iPhone SE 1st Gen and iPhone 12.

**Comment by gkpatel461** [comment]
@calda I have also faced the same issue for a few JSON in my mobile app. and I have also tried with give JSON by @chaudharyvikram then also getting the same issue. @calda Can please help to solve this issue? 

I want to know what the main issue Is it in an animation file or is it related to Mobile App or library?

**Comment by calda** [comment]
I took a look at the animation you shared @chaudharyvikram. The problem here is that the animation json itself is 20MB (very, very large) and contains over 650 images. When those images are decoded, they take up a very large amount of memory. Since this doesn't seem to be a bug in Lottie itself, I'm going to convert this from an issue to a discussion.

**Comment by chaudharyvikram** [comment]
So there is only one solution that need to reduce images or need to use external images?
